homeassistant/components/unifi/diagnostics.py

```diff
@@ -0,0 +1,99 @@
+"""Diagnostics support for UniFi Network."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from itertools import chain
+from typing import Any
+
+from homeassistant.components.diagnostics import REDACTED, async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import CONF_CONTROLLER, DOMAIN as UNIFI_DOMAIN
+
+TO_REDACT = {CONF_CONTROLLER, CONF_PASSWORD}
+REDACT_CLIENTS = {"bssid", "essid"}
+REDACT_DEVICES = {
+    "anon_id",
+    "gateway_mac",
+    "geo_info",
+    "serial",
+    "x_authkey",
+    "x_fingerprint",
+    "x_iapp_key",
+    "x_ssh_hostkey_fingerprint",
+    "x_vwirekey",
+}
+REDACT_WLANS = {"bc_filter_list", "x_passphrase"}
+
+
+@callback
+def async_replace_data(data: Mapping, to_replace: dict[str, str]) -> dict[str, Any]:
```

Comment:
why not using `async_redact_data`? This does the same ;)

Reply:
It does not. It replaces Mac addresses so it's still possible to view the topology

Reply:
ah I see now ... the mac addresses are not values, they are keys?

Reply:
On object store level they are keys in order to find the right object, but they are also spread out in many places, f.ex what switch client is connected to or what Mac a switch port has. You can see an example in the asserted dictionary in the test